tests/avro/test_AvroTransformer.py

```diff
@@ -0,0 +1,170 @@
+            "isReady": "isReady"
+        })
+
+    def test__create_transform_dict(self):
+        # It is complicated to test the values (functions) of the dict
+        transform_dict = AvroTransformer._create_transform_dict(self.example_avro_schema)
+        self.assertListEqual(list(transform_dict.keys()), ["startDate", "isReady"])
+
+    def test__create_defaults_dict(self):
+        defaults_dict = AvroTransformer._create_defaults_dict(self.example_avro_schema)
+        self.assertDictEqual(defaults_dict, {
+            "url": NoDefault,
+            "code": None,
+            "date": NoDefault,
+            "startDate": None,
+            "isReady": False
+        })
+
+    def test_transform_function_int2boolean(self):
+        int2boolean = AvroTransformer._get_transform_function("int2boolean")
+        self.assertEquals(int2boolean(0, {}), False)
+        self.assertEquals(int2boolean(-1, {}), False)
+        self.assertEquals(int2boolean(-30, {}), False)
+        self.assertEquals(int2boolean(1, {}), True)
+        self.assertEquals(int2boolean(2, {}), True)
+        self.assertEquals(int2boolean(30, {}), True)
+
+    def test_transform_function_copyFrom(self):
+        example_record_1 = {"word": "hola", "date": "yesterday", "num": 5}
+        example_record_2 = {"word": "hola", "num": 5}
+        copy_from_date = AvroTransformer._get_transform_function("copyFrom(date)")
+        self.assertEquals(copy_from_date("aux", example_record_1), "yesterday")
+        with self.assertRaises(RuntimeError) as context:
+            copy_from_date("aux", example_record_2)
+
+    def test_get_transformed_record(self):
+        example_record = {
+            "url": "http://google.com",
+            "code": "123456",
+            "date": "123456789",
+            "startDate": None,
+            "isReady": 1
+        }
+        avro_transformer = AvroTransformer(self.example_avro_schema)
+        self.assertDictEqual(
+            avro_transformer.get_transformed_record(example_record),
+            {
+                "url": "http://google.com",
+                "code": "123456",
+                "date": "123456789",
+                "startDate": "123456789",
+                "isReady": True
+            }
+        )
+
+    def test_get_renamed_record_and_remove_invalid_fields(self):
+        example_record = {
+            "url": "http://google.com",
+            "id": "123456",
+            "invalidField": ":-)",
+            "lastupdate": "123456789",
+            "isReady": 1
+        }
+        avro_transformer = AvroTransformer(self.example_avro_schema)
+        self.assertDictEqual(
+            avro_transformer.get_renamed_record_and_remove_invalid_fields(example_record),
+            {
+                "url": "http://google.com",
+                "code": "123456",
+                "date": "123456789",
+                "isReady": 1
+            }
+        )
+
+    def test_get_record_with_defaults(self):
```

Comment:
When there is an empty record, is it also completely filled with defaults?

Reply:
No, it isn't. It rise an error with the message `Required url not in record {}` 
In this example test, _url_ is a field with no default value, so it is automatically consider as a required field (also _date_)

Reply:
Ok, thanks for the explanation!